File: backend/app/models/pwd_reset_req.py
from typing import Optional

from pydantic import EmailStr, constr

from app.models.core import CoreModel, DateTimeModelMixin, IDModelMixin

# ? until constr fixed
# mypy: ignore-errors


class PasswordResetRequest(CoreModel, DateTimeModelMixin, IDModelMixin):
    """
    Users can request a password reset to an administrator.
    """

    email: EmailStr
    message: constr(min_length=2)


class PasswordResetRequestCreate(CoreModel):
    """
    Users can request a password reset to an administrator.
    """

    email: EmailStr
    message: constr(min_length=2)

File: backend-go/internal/app/models/pwd_reset_req.go
package models

type PasswordResetRequest struct {
	CoreModel
	DateTimeModelMixin
	IDModelMixin
	Email   string `validate:"required,email"`
	Message string `validate:"required,min=2"`
}

type PasswordResetRequestCreate struct {
	Email   string `validate:"required,email"`
	Message string `validate:"required,min=2"`
}
